src/IntegrationsGoeController.cpp

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * Copyright (C) 2023-2024 Thomas Basler and others
+ */
+#include "IntegrationsGoeController.h"
+#include "Configuration.h"
+#include "Datastore.h"
+#include "MessageOutput.h"
+#include "NetworkSettings.h"
+#include <Hoymiles.h>
+
+IntegrationsGoeControllerClass IntegrationsGoeController;
+
+IntegrationsGoeControllerClass::IntegrationsGoeControllerClass()
+    : _loopTask(TASK_IMMEDIATE, TASK_FOREVER, std::bind(&IntegrationsGoeControllerClass::loop, this))
+{
+}
+
+void IntegrationsGoeControllerClass::init(Scheduler& scheduler)
+{
+    using std::placeholders::_1;
+
+    NetworkSettings.onEvent(std::bind(&IntegrationsGoeControllerClass::NetworkEvent, this, _1));
+
+    scheduler.addTask(_loopTask);
+    _loopTask.setInterval(Configuration.get().Integrations.GoeControllerUpdateInterval * TASK_SECOND);
+    _loopTask.enable();
+}
+
+void IntegrationsGoeControllerClass::NetworkEvent(network_event event)
+{
+    switch (event) {
+    case network_event::NETWORK_GOT_IP:
+        _networkConnected = true;
+        break;
+    case network_event::NETWORK_DISCONNECTED:
+        _networkConnected = false;
+        break;
+    default:
+        break;
+    }
+}
+
+void IntegrationsGoeControllerClass::loop()
+{
+    const auto& integrationsConfig = Configuration.get().Integrations;
+
+    const bool reachable = Datastore.getIsAllEnabledReachable();
+
+    _loopTask.setInterval((reachable ? integrationsConfig.GoeControllerUpdateInterval : std::min(integrationsConfig.GoeControllerUpdateInterval, 5U)) * TASK_SECOND);
```

Comment:
Increase timeout when no data is available because we do not need to publish "0" so often.

The go-e Controller needs to be fed a value every couple seconds or else it will "forget" the value was set, so not sending does not work unfortunately. I also do not know the exact interval tho.